File: coders/curso_python/poo/desafio_carro/carro.py
class Carro:
    def __init__(self, velocidade_maxima):
        self.velocidade_maxima = velocidade_maxima
        self.velocidade_atual = 0

    def acelera(self, delta=5):
        nova_velocidade = self.velocidade_atual + delta

        self.velocidade_atual = nova_velocidade \
            if nova_velocidade < self.velocidade_maxima \
            else self.velocidade_maxima

        return self.velocidade_atual

    def freia(self, delta=5):
        nova_velocidade = self.velocidade_atual - delta

        self.velocidade_atual = nova_velocidade \
            if nova_velocidade > 0 \
            else 0

        return self.velocidade_atual

File: coders/kotlin/Exemplos/src/oo/heranca/Carro.kt
package oo.heranca

open class Carro(val velocidadeMaxima: Int, var velocidadeAtual: Int = 0) {

    protected fun alterarVelocidade(novaVelocidade: Int) {
        velocidadeAtual = when (novaVelocidade) {
            in 0..velocidadeMaxima -> novaVelocidade
            in velocidadeMaxima + 1..Int.MAX_VALUE -> velocidadeMaxima
            else -> 0
        }
    }

    open fun acelerar() {
        alterarVelocidade(velocidadeAtual + 5)
    }

    open fun freiar() {
        alterarVelocidade(velocidadeAtual - 5)
    }
}